jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -24,9 +24,58 @@
 from jdaviz.core.tools import ICON_DIR
 from jdaviz.core.custom_traitlets import IntHandleEmpty
 
+from scipy.interpolate import make_interp_spline
+
 __all__ = ['PlotOptions']
 
 
+class SplineStretch:
+    """
+    A class to represent spline stretches.
+
+    Attributes:
+        k (int): Degree of the smoothing spline.
+        bc_type (str or None): Boundary condition type.
+        t (array-like or None): Array of knot positions.
+        x (array-like): The x-coordinates of the data points.
+        y (array-like): The y-coordinates of the data points.
+        spline (object): Interpolating spline.
+    """
+
+    def __init__(self):
+        """
+        Initializes the SplineStretch class with default knot values.
+        """
+        self.k = 3

```

Comment:
Please explain why "3" is a good default.

Reply:
We opted for a cubic spline (degree 3) because it provides a smooth curve without overfitting. Also confirmed with POs that a cubic spline is suitable for our needs. As we consider making knots editable in the future, we can revisit and discuss the flexibility of supporting different spline degrees.

Reply:
Can you please put that as inline comment (instead of PR comment) for future reference? Thanks!